lib/LedgerSMB/Middleware/AuthenticateSession.pm

```diff
@@ -154,26 +136,20 @@ sub call {
         my $version =
             LedgerSMB::DBH->require_version($dbh, $LedgerSMB::VERSION);
         if ($version) {
+            $env->{'lsmb.session.expire'} = 1;
             return LedgerSMB::PSGI::Util::incompatible_database(
                 $LedgerSMB::VERSION, $version);
         }
 
-        $extended_cookie = _verify_session($env->{'lsmb.db'},
-                                           $env->{'lsmb.company'},
-                                           $session);
-        if (! $extended_cookie) {
+        if (! _verify_session($env->{'lsmb.db'},
+                              $env->{'lsmb.company'},
+                              $session)) {
             $dbh->commit;  # potentially log something
```

Comment:
There will always be something to commit?

Reply:
I'm not sure. I think at least sometimes: when the session is expired, we want to commit the removal of the session.

Reply:
Will we generate an error if there’s nothing to commit?

Reply:
Nope. it just commits an empty transaction.